api/proto/teleport/notifications/v1/notifications_service.proto

```diff
@@ -31,31 +31,30 @@ service NotificationService {
   rpc CreateUserNotification(CreateUserNotificationRequest) returns (Notification);
   // DeleteUserNotification deletes a user-specific notification.
   rpc DeleteUserNotification(DeleteUserNotificationRequest) returns (google.protobuf.Empty);
-  // ListUserNotifications returns a page of user-specific notifications.
-  rpc ListUserNotifications(ListUserNotificationsRequest) returns (ListUserNotificationsResponse);
+  // DeleteAllUserNotificationsForUser deletes all of a user's user-specific notifications.
+  rpc DeleteAllUserNotificationsForUser(DeleteAllUserNotificationsForUserRequest) returns (google.protobuf.Empty);
```

Comment:
When a local user is deleted, I think it makes sense for us to delete all of their user-specific notifications to prevent a bunch of notifications from accumulating in the backend for users that don't exist anymore and that serve no purpose.